File: app/src/main/java/com/akilimo/mobile/views/fragments/InfoFragment.java
package com.akilimo.mobile.views.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.akilimo.mobile.databinding.FragmentInfoBinding;
import com.akilimo.mobile.inherit.BaseStepFragment;
import com.stepstone.stepper.VerificationError;

/**
 * A simple {@link Fragment} subclass.
 */
public class InfoFragment extends BaseStepFragment {

    FragmentInfoBinding binding;

    public InfoFragment() {
        // Required empty public constructor
    }

    public static InfoFragment newInstance() {
        return new InfoFragment();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        this.context = context;
    }

    @Override
    protected View loadFragmentLayout(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentInfoBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }


    @Nullable
    @Override
    public VerificationError verifyStep() {
        return null;
    }

    @Override
    public void onSelected() {

    }

    @Override
    public void onError(@NonNull VerificationError error) {

    }
}

File: app/src/main/java/com/akilimo/mobile/views/fragments/InfoFragment.kt
package com.akilimo.mobile.views.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.CompoundButton
import com.akilimo.mobile.R
import com.akilimo.mobile.databinding.FragmentInfoBinding
import com.akilimo.mobile.inherit.BindBaseStepFragment
import com.stepstone.stepper.VerificationError

class InfoFragment : BindBaseStepFragment<FragmentInfoBinding>() {

    companion object {
        fun newInstance(): InfoFragment {
            return InfoFragment()
        }
    }

    override fun inflateBinding(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ) = FragmentInfoBinding.inflate(inflater, container, false)

    override fun onBindingReady(savedInstanceState: Bundle?) {
        binding.chkAgreeTerms.setOnCheckedChangeListener { _: CompoundButton?, checked: Boolean ->
            sessionManager.setDisclaimerRead(checked)
        }
    }

    override fun verifyStep(): VerificationError? {
        if (!sessionManager.getDisclaimerRead()) {
            return VerificationError(getString(R.string.lbl_agree_to_disclaimer))
        }
        return null
    }
}
